### DISCUSSION TITLE
Various network interfaces across multiple devices

### DISCUSSION THREAD

**Discussion by irish1986** [techno-tim/k3s-ansible#21] [Questions]
I am not sure this is the core of my issues yet I am trying to build a multi-arch cluster split between Ubuntu VMs and RPi4.  Ubuntu VM network interface are `ens18` and RPi4 are `eth0` which obviously had me ran into a few issues.  Is there an easy workaround ? Where could I find documentation around this items ?

**Comment by Jasmin25** [comment]
Kind on a similar page, so I connected all my machines through tailscale. And using `flannel_iface: "tailscale0"` in all.yml file. Still the nodes fail to connect to each other, even though connectivity between nodes on tailscale network interface is fine. 

The systemctl status on k3s-node is throwing this error:

```
level=error msg="Failed to connect to proxy" error="dial tcp 10.0.0.33:6443: connect: no route to host"
level=error msg="Remotedialer proxy error" error="dial tcp 10.0.0.33:6443: connect: no route to host"
```
10.0.0.33 IP is from default interface, not tailscale :/

**Reply by irish1986** [reply]
Back to this thread after a few days, I am not sure where to start this investigation. Suggestions? Have you tried setting this with K3Sup? I was in the middle of trying out that approach but my girlfriend went into labor and now I have a daughter.

**Reply by Jasmin25** [reply]
Well Congrats man!

**Reply by Jasmin25** [reply]
Thanks for recommending k3sup, will check it out.

**Comment by ColdCanuck** [comment]
Sorry late to the party. How is life as the sleep deprived ?

You can easily change your ubuntu VMs to be on "eth0".  Add a udev rule

cat  /etc/udev/rules.d/70-persistent-net.rules 
SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", ATTR{address}=="fe:ea:a2:27:07:fc", NAME="eth0"

Adjust for your mac address. This seems to happen automagically on Debian cloud-init VMs.

**Comment by Ashmodaj** [comment]
Hi,

Had the same issue with unmatching network interface name while mixing Elitedesk mini, RPI and other odds systems.
The simpliest solution I found was to simply rename the interfaces using netplan.

Edit the /etc/netplan/*.yml adding the matching mac address of your adapter and the set_name argument with the desired name.

```
network:
    version: 2
    ethernets:
        eno1:
            dhcp4: true
            match:
                macaddress: <YOUR MAC ID HERE>
            set-name: eth0
```
Then

`netplan try`
`netplan apply`

It's not beautiful but it work.
